lib/msf/util/exe.rb

```diff
@@ -854,6 +854,12 @@ def self.to_osx_app(exe, opts = {})
     hidden      = opts.fetch(:hidden, true)
     plist_extra = opts.fetch(:plist_extra, '')
 
+    # The app won't execute if the macho isn't executable
+    macho = File.new(exe_name, 'w')
+    macho.write(exe)
+    macho.close
+    FileUtils.chmod(777, macho)
```

Comment:
This doesn't actually do anything. It's just creating a local file, settings permission on it, then deleting it later.